lib/kafka/async_producer.rb

```diff
@@ -274,6 +294,7 @@ def produce(value, **kwargs)
             sleep @retry_backoff**retries
             retry
           else
+            @finally.call(value)
```

Comment:
OK just realized here is an issue. I thought it'd be multiple messages but the call to producer above (`@producer.produce(value, **kwargs)`) which seems to take something like `produce(message, {"topic" => "foo.whatever"..... etc})` so I'm trying to think of how we should recover rom here (i.e. package the message back up into the `[Message]` type that comes off the queue.